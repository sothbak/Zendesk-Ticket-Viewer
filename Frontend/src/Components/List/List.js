import React, { Component } from 'react';
import Ticket from '../Ticket/Ticket';
import styles from './List.module.css';

// array of Tickets
class List extends Component {
    render() {
        let list_tickets = this.props.ticket_data.map(ticket => {
            return ( 
                <Ticket 
                    created_at={ticket.created_at}
                    description={ticket.description}
                    key={ticket.id}
                    id={ticket.id}
                    subject={ticket.subject}
                    clickDisplayTicket={this.props.clickDisplayTicket} /> )
        });

        return (
            <ul className={styles.list}>
                {list_tickets}
            </ul>
        )
    }
}

export default List;