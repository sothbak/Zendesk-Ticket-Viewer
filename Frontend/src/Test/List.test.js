import React from 'react';
import { shallow } from 'enzyme';
import List from '../Components/List/List.js';
describe('Ticket', () => {
  it('should render correctly', () => {
    const props = [{created_at:"created_at", description:"description", id:"id", subject:"subject"}];
    const wrapper = shallow(<List ticket_data={props} />);
    expect(wrapper).toMatchSnapshot();
  });

});