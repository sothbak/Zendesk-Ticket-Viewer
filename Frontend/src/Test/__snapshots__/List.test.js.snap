// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly 1`] = `
<ul
  className="list"
>
  <Tickets
    created_at="2021-05-04T01:05:53Z"
    description="Hi Sothea-Roth,

This is your first ticket. Ta-da! Any customer request sent to your supported channels (email, chat, voicemail, web form, and tweet) will become a Support ticket, just like this one. Respond to this ticket by typing a message and clicking Submit. You can also see how an email becomes a ticket by emailing your new account, support@sothearoth.zendesk.com. Your ticket will appear in ticket views.

That's the ticket on tickets. If you want to learn more, check out: 
https://support.zendesk.com/hc/en-us/articles/203691476
"
    id={1}
    key="1"
    subject="Sample ticket: Meet the ticket"
  />
  <Tickets
    created_at="2021-05-04T01:07:29Z"
    description="Hello,
Something dramatic happened and I could really use your help.
Thanks in advance,
Customer
"
    id={2}
    key="2"
    subject="I need help"
  />
  <Tickets
    created_at="2021-05-04T01:57:42Z"
    description="Aute ex sunt culpa ex ea esse sint cupidatat aliqua ex consequat sit reprehenderit. Velit labore proident quis culpa ad duis adipisicing laboris voluptate velit incididunt minim consequat nulla. Laboris adipisicing reprehenderit minim tempor officia ullamco occaecat ut laborum.

Aliquip velit adipisicing exercitation irure aliqua qui. Commodo eu laborum cillum nostrud eu. Mollit duis qui non ea deserunt est est et officia ut excepteur Lorem pariatur deserunt."
    id={3}
    key="3"
    subject="velit eiusmod reprehenderit officia cupidatat"
  />
  <Tickets
    created_at="2021-05-04T01:57:42Z"
    description="Exercitation amet in laborum minim. Nulla et veniam laboris dolore fugiat aliqua et sit mollit. Dolor proident nulla mollit culpa in officia pariatur officia magna eu commodo duis.

Aliqua reprehenderit aute qui voluptate dolor deserunt enim aute tempor ad dolor fugiat. Mollit aliquip elit aliqua eiusmod. Ex et anim non exercitation consequat elit dolore excepteur. Aliqua reprehenderit non culpa sit consequat cupidatat elit."
    id={4}
    key="4"
    subject="excepteur laborum ex occaecat Lorem"
  />
  <Tickets
    created_at="2021-05-04T01:57:42Z"
    description="Sunt incididunt officia proident elit anim ullamco reprehenderit ut. Aliqua sint amet aliquip cillum minim magna consequat excepteur fugiat exercitation est exercitation. Adipisicing occaecat nisi aliqua exercitation.

Aute Lorem aute tempor sunt mollit dolor in consequat non cillum irure reprehenderit. Nulla deserunt qui aliquip officia duis incididunt et est velit nulla irure in fugiat in. Deserunt proident est in dolore culpa mollit exercitation ea anim consequat incididunt. Mollit et occaecat ullamco ut id incididunt laboris occaecat qui."
    id={5}
    key="5"
    subject="ad sunt qui aute ullamco"
  />
  <Tickets
    created_at="2021-05-04T01:57:43Z"
    description="Pariatur voluptate laborum voluptate sunt ad magna exercitation nulla. In in Lorem minim dolor laboris reprehenderit ad Lorem. Cupidatat laborum qui mollit nostrud magna ullamco. Tempor nisi ex ipsum fugiat dolor proident qui consectetur anim sunt. Dolore consectetur in ex esse et aliqua fugiat enim Lorem ea Lorem incididunt. Non amet elit pariatur commodo labore officia esse anim. In do mollit commodo nulla ullamco culpa cillum incididunt.

Et nostrud aute fugiat voluptate tempor ad labore in elit et sunt. Enim quis nulla eu ut sit. Pariatur irure officia occaecat non dolor est excepteur anim incididunt commodo ea cupidatat minim excepteur. Cillum proident tempor laborum amet est ipsum ipsum aliqua sit sunt reprehenderit fugiat aliqua ea."
    id={6}
    key="6"
    subject="aliquip mollit quis laborum incididunt"
  />
  <Tickets
    created_at="2021-05-04T01:57:43Z"
    description="Aute Lorem fugiat ad consectetur sint fugiat. Et qui ipsum in qui nostrud nulla qui et occaecat culpa ad occaecat. Aute mollit occaecat mollit proident nostrud non ea laboris adipisicing deserunt officia. Eiusmod sint fugiat veniam consectetur consequat exercitation esse est.

Velit est ipsum labore aliquip quis mollit laborum in. Consectetur cillum proident ullamco exercitation pariatur. Incididunt consectetur tempor adipisicing esse minim mollit Lorem."
    id={7}
    key="7"
    subject="nisi aliquip ipsum nostrud amet"
  />
  <Tickets
    created_at="2021-05-04T01:57:44Z"
    description="Deserunt officia ea fugiat dolor eu laboris esse reprehenderit deserunt dolore voluptate amet culpa. Proident ut mollit adipisicing occaecat Lorem do ut ex aute laboris fugiat minim dolor. In anim non nostrud adipisicing aliquip nisi laborum cupidatat officia. Sunt cillum sint anim elit culpa commodo amet excepteur consectetur veniam nulla ut. Officia ut ut sit incididunt cupidatat velit proident cupidatat voluptate eu ex.

Velit nisi voluptate nulla reprehenderit officia consectetur dolor nostrud cillum duis. Dolore cupidatat eu veniam ut fugiat mollit ea. Reprehenderit nulla nisi cillum voluptate ex. Occaecat incididunt id mollit deserunt occaecat adipisicing ullamco ipsum. Minim ullamco adipisicing quis aliquip est ex sunt ea quis. Sint aute occaecat velit ipsum enim qui fugiat esse in officia excepteur irure. Enim eu dolore reprehenderit exercitation ullamco est nulla voluptate consectetur aliqua sit."
    id={8}
    key="8"
    subject="cillum quis nostrud labore amet"
  />
  <Tickets
    created_at="2021-05-04T01:57:44Z"
    description="Aute mollit ex nulla id culpa nisi enim amet enim duis laborum in. Eiusmod ea quis tempor voluptate deserunt tempor amet fugiat esse ea. Anim sunt nisi laboris dolor. Est ullamco voluptate dolor magna elit eu. Laboris reprehenderit irure occaecat excepteur.

Do nisi commodo dolor eiusmod eiusmod non eu occaecat sunt incididunt irure consectetur do. Exercitation occaecat culpa voluptate exercitation sit. In aute eu nulla anim reprehenderit velit. Eu deserunt aliquip aute Lorem consectetur mollit non esse veniam irure excepteur cupidatat excepteur officia. Velit id et mollit cupidatat cillum elit dolor fugiat pariatur. Deserunt non eu consectetur duis ex consectetur duis eu proident."
    id={9}
    key="9"
    subject="proident est nisi non irure"
  />
  <Tickets
    created_at="2021-05-04T01:57:45Z"
    description="Ex non officia in ullamco veniam fugiat cupidatat id aute. Quis minim et quis laborum excepteur. Non officia quis tempor quis nisi et. Aliqua tempor voluptate nulla esse sint. Id nulla consequat sint eiusmod nisi.

Cupidatat anim magna qui aliquip. Anim deserunt sint incididunt labore aliquip. Reprehenderit magna deserunt reprehenderit irure Lorem nulla est officia cupidatat in. Dolore ullamco veniam proident consectetur amet ad nulla amet commodo enim occaecat. Sint fugiat dolor aliqua proident. Ex enim eu pariatur qui officia cupidatat adipisicing esse qui fugiat. Do mollit quis cillum quis qui occaecat labore."
    id={10}
    key="10"
    subject="veniam ea eu minim aute"
  />
  <Tickets
    created_at="2021-05-04T01:57:45Z"
    description="Sit sit consequat magna aliquip officia qui. Fugiat amet id dolor sint exercitation sit. Eiusmod ex eiusmod voluptate voluptate est amet non culpa minim enim minim. Eiusmod fugiat veniam duis eiusmod sint laborum ex amet occaecat.

Nostrud consequat officia tempor amet eu. Non adipisicing dolore amet minim id consequat labore irure in esse et aliqua pariatur. Aliquip aliqua id ipsum amet laboris exercitation sunt cillum est et est. Tempor amet qui do dolore fugiat ad id nulla ullamco dolore tempor irure deserunt magna. Ipsum voluptate aliquip ut ad in pariatur adipisicing occaecat ea excepteur Lorem enim exercitation. Lorem sunt officia voluptate pariatur labore esse nostrud ullamco irure sit. Voluptate exercitation do aliquip eu consectetur."
    id={11}
    key="11"
    subject="magna reprehenderit nisi est cillum"
  />
  <Tickets
    created_at="2021-05-04T01:57:46Z"
    description="Magna culpa velit duis nisi sint veniam nisi adipisicing aute. Eiusmod id cupidatat pariatur tempor esse do. Aliquip ipsum voluptate eiusmod mollit. Et ullamco consectetur tempor cillum sunt anim enim cillum reprehenderit cillum proident. Eu aliqua excepteur eiusmod non ullamco aliqua esse laboris ad commodo reprehenderit pariatur consequat est.

Voluptate eiusmod officia culpa nulla culpa mollit reprehenderit cillum adipisicing sit consectetur in. Anim occaecat excepteur commodo officia aute voluptate excepteur do ut esse occaecat duis consequat. Sunt minim est in non exercitation ad minim ullamco dolore in enim mollit aliquip. Ut ex dolore proident Lorem ut dolor ex. Laboris elit in in mollit. Quis id minim ex ad qui velit."
    id={12}
    key="12"
    subject="quis veniam ad sunt non"
  />
  <Tickets
    created_at="2021-05-04T01:57:46Z"
    description="Amet sint ea minim excepteur amet. Eiusmod duis ad ea ad aliquip occaecat ea voluptate sunt cillum duis aute. Esse in duis tempor et. Est duis nulla irure ullamco amet sit esse Lorem eu aliqua ullamco sunt. In sint dolore id eu ea.

Irure proident nulla deserunt officia laboris culpa. Qui est adipisicing elit tempor cupidatat minim laborum ea est nostrud nulla mollit id dolore. Aute amet amet deserunt incididunt magna ipsum exercitation nostrud incididunt ea qui anim. Adipisicing irure commodo consectetur Lorem ad excepteur nulla eiusmod aliquip irure occaecat. Id sint fugiat proident Lorem est minim ea sint irure. Veniam sint ex minim duis est."
    id={13}
    key="13"
    subject="tempor aliquip sint dolore incididunt"
  />
  <Tickets
    created_at="2021-05-04T01:57:46Z"
    description="Quis veniam consectetur ullamco aute incididunt eiusmod ut consectetur sunt laboris sint officia labore. Incididunt sit elit ipsum esse excepteur veniam Lorem excepteur amet amet tempor officia. Magna reprehenderit voluptate elit adipisicing consectetur laboris eu nisi ad excepteur adipisicing non duis proident.

Aliqua sunt ea dolor dolor elit occaecat. Aliqua reprehenderit commodo minim eu exercitation incididunt consectetur. Eiusmod anim ad adipisicing et ullamco id incididunt."
    id={14}
    key="14"
    subject="labore pariatur ut laboris laboris"
  />
  <Tickets
    created_at="2021-05-04T01:57:47Z"
    description="Fugiat magna dolor ipsum dolor ex qui commodo deserunt veniam aliqua qui ex. Consequat est duis velit eu id labore id dolor sint laboris. Minim nisi sint aliquip nisi cillum magna deserunt non cupidatat est. Exercitation ea pariatur ipsum non pariatur dolor velit laboris dolore irure reprehenderit ut.

Amet et officia eiusmod adipisicing. Amet ut ut tempor cillum aliqua elit mollit mollit. Exercitation commodo culpa adipisicing aute voluptate consectetur ut est ea quis aliquip duis excepteur. Anim quis quis occaecat fugiat occaecat. Non et deserunt anim eu anim non laborum ex tempor. Tempor aute reprehenderit minim consequat elit minim. Aliqua in laboris culpa excepteur qui enim excepteur cillum commodo eiusmod eu enim do ullamco."
    id={15}
    key="15"
    subject="officia mollit aliqua eu nostrud"
  />
  <Tickets
    created_at="2021-05-04T01:57:47Z"
    description="Lorem sit mollit qui ad culpa do commodo ipsum. Labore elit cillum irure mollit est est cupidatat consectetur commodo laborum qui cupidatat. Ullamco minim veniam est laboris aliquip consectetur voluptate.

Sunt anim aute culpa mollit culpa veniam eiusmod sunt proident. Ea pariatur esse occaecat aute exercitation quis ut quis ullamco nisi exercitation non commodo. Voluptate mollit consequat velit veniam eiusmod culpa proident occaecat. Labore exercitation enim culpa eu labore proident exercitation ex culpa Lorem deserunt nulla nostrud."
    id={16}
    key="16"
    subject="do incididunt incididunt quis anim"
  />
  <Tickets
    created_at="2021-05-04T01:57:48Z"
    description="Anim laborum deserunt cillum minim elit commodo sunt ullamco sint laborum exercitation sunt eiusmod commodo. Proident labore culpa anim elit culpa reprehenderit. Ullamco aute cillum dolor sit quis ullamco. Esse labore labore velit dolor. Velit voluptate proident amet incididunt cillum incididunt aliquip commodo quis officia.

Irure non laborum in enim laborum dolore irure amet anim irure adipisicing. Ex culpa fugiat commodo quis sunt Lorem sit commodo ullamco pariatur sit proident. Enim esse voluptate labore consequat ipsum cupidatat elit anim nisi minim."
    id={17}
    key="17"
    subject="tempor magna anim ea id"
  />
  <Tickets
    created_at="2021-05-04T01:57:48Z"
    description="Qui magna exercitation labore occaecat dolor duis officia pariatur do amet nulla laboris ex. Amet qui laboris esse non quis. Eu anim ex sunt ut. Veniam nulla amet ea pariatur consequat elit fugiat nulla minim voluptate tempor sunt et sit. Esse exercitation officia aliqua labore eiusmod. Cillum duis veniam voluptate elit quis sunt commodo sunt. Nulla consectetur id laboris aute laborum culpa exercitation sit commodo.

In officia pariatur anim in consequat ex. Do magna labore esse elit sit est excepteur aliquip est sint. Nulla aute aliquip officia cillum tempor incididunt voluptate proident mollit dolore ullamco. Deserunt adipisicing cillum in ea Lorem elit."
    id={18}
    key="18"
    subject="exercitation sit incididunt magna laboris"
  />
  <Tickets
    created_at="2021-05-04T01:57:48Z"
    description="Id commodo nisi velit irure et ad occaecat. Labore magna ex sint ut commodo ullamco et. Non aliqua aliquip et do irure. Consequat consectetur sit cupidatat est non laboris. Id pariatur ipsum aliqua sunt labore proident cillum occaecat mollit.

Nulla mollit anim ea culpa tempor id aliqua quis sit dolor esse ipsum. Adipisicing nisi irure mollit voluptate nisi ut laboris cillum velit qui. Incididunt excepteur sint magna eiusmod minim adipisicing consectetur consequat deserunt commodo et pariatur."
    id={19}
    key="19"
    subject="laborum ea ut in cupidatat"
  />
  <Tickets
    created_at="2021-05-04T01:57:49Z"
    description="Tempor id commodo ut eu. Ipsum ut velit deserunt ad minim ullamco aliquip eu eiusmod veniam. Elit nostrud Lorem aute nisi commodo in aliquip elit nisi ea. Magna mollit incididunt consequat elit proident adipisicing et anim enim. Sint ipsum ullamco duis tempor nostrud veniam. Anim quis irure ullamco reprehenderit nostrud id reprehenderit anim. Nulla id do ullamco nisi.

Commodo laborum excepteur ad ut ipsum fugiat tempor nostrud officia. Consequat non adipisicing et anim dolore magna aliquip cillum Lorem. Pariatur veniam do esse magna excepteur ex officia quis sit amet ut in et amet. Ipsum est dolore do consectetur ut mollit proident sit aute labore veniam nulla velit anim."
    id={20}
    key="20"
    subject="est fugiat labore pariatur esse"
  />
  <Tickets
    created_at="2021-05-04T01:57:49Z"
    description="Enim qui elit minim cillum qui nisi deserunt ex excepteur nulla ex sint ut adipisicing. Lorem labore nulla non id aliquip ex excepteur est excepteur. Eu reprehenderit culpa consequat voluptate ullamco aute consequat.

Reprehenderit laborum deserunt minim exercitation anim officia ullamco duis anim. Officia adipisicing cillum aliquip exercitation do. Deserunt velit aute excepteur sit elit consequat reprehenderit occaecat nostrud quis consectetur ut. Voluptate mollit reprehenderit veniam qui cillum duis commodo exercitation enim cupidatat sunt voluptate velit non. Id pariatur aliqua in ipsum anim culpa non consectetur occaecat ut. Ex ex adipisicing ut sint mollit nisi consequat aute excepteur."
    id={21}
    key="21"
    subject="commodo sint laboris est et"
  />
  <Tickets
    created_at="2021-05-04T01:57:50Z"
    description="Occaecat eu exercitation deserunt proident occaecat officia esse commodo aliqua pariatur dolor tempor sit. Nisi est eu excepteur ullamco in. Irure ut reprehenderit dolore tempor do laboris voluptate minim aliqua cupidatat.

Sint dolore veniam sit fugiat elit aute proident adipisicing laborum deserunt aliquip officia magna. Fugiat in nisi aliquip occaecat duis commodo id. Sit tempor voluptate ullamco labore nostrud enim enim tempor mollit reprehenderit tempor ea. Adipisicing ex ullamco nulla cillum excepteur incididunt aliquip mollit ullamco occaecat. In excepteur deserunt reprehenderit ut adipisicing ad. Adipisicing ea esse officia est. Cillum mollit dolore incididunt est qui adipisicing ea irure sint cillum nostrud ad."
    id={22}
    key="22"
    subject="laboris sint Lorem ex Lorem"
  />
  <Tickets
    created_at="2021-05-04T01:57:50Z"
    description="Anim occaecat nisi et velit fugiat tempor incididunt cillum nostrud ut sunt cupidatat ipsum. Tempor adipisicing elit consectetur culpa. Ad dolor velit consectetur tempor commodo est ipsum cillum incididunt esse ad ex velit. Proident eu esse eiusmod nulla mollit eu cupidatat consectetur consectetur exercitation reprehenderit laborum adipisicing do.

Lorem ad aliquip veniam nisi exercitation sit. Fugiat reprehenderit eu do officia commodo ea cillum et sunt laborum nostrud in. Dolore eu ad qui eu labore qui pariatur id. Et enim amet magna quis mollit qui. Esse laboris id voluptate do. Sunt mollit dolore proident fugiat minim mollit culpa esse minim sint ut consectetur occaecat. Incididunt culpa non nulla veniam ea laborum aliqua tempor nulla ea."
    id={23}
    key="23"
    subject="esse adipisicing consectetur sunt tempor"
  />
  <Tickets
    created_at="2021-05-04T01:57:51Z"
    description="Ut officia ut elit non velit nisi quis adipisicing consequat nostrud occaecat cillum. Do exercitation nulla officia consequat est velit eu id in et. Nulla culpa ullamco minim aliquip nisi ad ex. Qui aliquip ut esse ipsum magna do ipsum esse voluptate magna nisi sint ex. Laborum commodo culpa laboris quis incididunt. In enim dolore ad sunt aliquip nulla aliqua aute et amet ut veniam labore. Consectetur eiusmod et aliquip proident est Lorem proident nostrud esse eu consequat excepteur.

Exercitation duis labore irure deserunt reprehenderit do non. Incididunt dolore do nisi consectetur mollit anim in. Do dolor dolor amet enim et quis voluptate elit magna non laboris. Enim enim ea occaecat minim mollit velit id exercitation ea officia incididunt proident. Dolore fugiat esse Lorem amet consequat magna laborum fugiat. Id aliqua labore consequat laborum cillum fugiat incididunt non do ipsum."
    id={24}
    key="24"
    subject="sunt enim pariatur id id"
  />
  <Tickets
    created_at="2021-05-04T01:57:51Z"
    description="Consectetur dolore in minim anim reprehenderit do labore commodo officia ad ipsum amet quis ipsum. Amet ad dolor minim sint excepteur adipisicing exercitation elit. Non aliquip commodo voluptate sint veniam et ipsum magna occaecat do consequat incididunt magna do. Pariatur proident aliquip veniam dolore occaecat voluptate excepteur fugiat consequat in. Dolor ullamco ea cupidatat nostrud. Deserunt deserunt aliqua elit aute qui elit et in. Mollit cupidatat culpa officia ad qui deserunt tempor sint reprehenderit proident consectetur amet aliquip.

Adipisicing sunt consequat labore ad ut elit anim minim esse Lorem non qui consequat. Dolore anim adipisicing ullamco nisi anim ipsum nulla magna proident nisi nulla ea. Ullamco duis nisi aliquip pariatur nulla ad occaecat."
    id={25}
    key="25"
    subject="et ad ut enim labore"
  />
</ul>
`;
